src/Conseil.tsx

```diff
@@ -0,0 +1,75 @@
+import * as querystring from 'querystring';
+
+export interface ConseilFilter {
+    limit: number;
+    block_id: string[];
+    block_level: number[];
+    block_netid: string[];
+    block_protocol: string[];
+    operation_id: string[];
+    operation_source: string[];
+    account_id: string[];
+    account_manager: string[];
+    account_delegate: string[];
+}
+
+function queryNautilus(network: string, command: string, payload: string): Promise<string> {
+    return new Promise((resolve, reject) => {
+        let url = `https://localhost:1337/tezos/${network}/${command}`;
+        const xmlHttp = new XMLHttpRequest();
+        xmlHttp.open('GET', url, true);
+        xmlHttp.setRequestHeader('apiKey', 'hooman');
+        xmlHttp.onload = () => resolve(xmlHttp.responseText);
+        xmlHttp.onerror = () => reject(xmlHttp.statusText);
+        xmlHttp.send(payload);
+    });
+}
+
+function sanitizeFilter(filter: ConseilFilter): ConseilFilter {
+    return {
+        limit: filter.limit,
+        block_id: filter.block_id.filter(Boolean),
+        block_level: filter.block_level.filter(Boolean),
+        block_netid: filter.block_netid.filter(Boolean),
+        block_protocol: filter.block_protocol.filter(Boolean),
+        operation_id: filter.operation_id.filter(Boolean),
+        operation_source: filter.operation_source.filter(Boolean),
+        account_id: filter.account_id.filter(Boolean),
+        account_manager: filter.account_manager.filter(Boolean),
+        account_delegate: filter.account_delegate.filter(Boolean)
+    };
+}
+
+function queryNautilusWithFilter(network: string, command: string, filter: ConseilFilter) {
+    let params = querystring.stringify(sanitizeFilter(filter));
+    let cmdWithParams = `${command}?${params}`;
+    return queryNautilus(network, cmdWithParams, '');
```

Comment:
Could be useful to wrap these async calls with some error handling. Then whenever you catch an error, you could provide some useful message to the user informing them

Reply:
Opened ticket #18 to do this.